src/kiara_plugin/topic_modelling/modules/corpus_preparation.py

```diff
@@ -109,4 +110,71 @@ def get_date(file):
         except Exception as e:
             print(f"An error occurred: {e}")
 
-        outputs.set_value("corpus_table", output_table)
\ No newline at end of file
+        outputs.set_value("corpus_table", output_table)
+
+class CorpusDistTime(KiaraModule):
+    """
+    This module aggregates a table by day, month or year from a corpus table that contains a date column. It returns the distribution over time, which can be used for display purposes, such as visualization.
+
+    Dependencies:
+    - polars: https://www.pola.rs/
+    - pyarrow: https://arrow.apache.org/docs/python/
+    - duckdb: https://duckdb.org/
+    
+    """
+
+    _module_type_name = "topic_modelling.time_dist"
+
+    def create_inputs_schema(self):
+        
+        return {
+             "periodicity": {
+                "type": "string",
+                "doc": "The desired data periodicity to aggregate the data. Values can be either 'day','month' or 'year'."
+            },
+            "date_col_name": {
+                "type": "string",
+                "doc": "The column that contains the date, a list of compliant formats can be consulted here: https://docs.rs/chrono/latest/chrono/format/strftime/index.html."
+            },
+            "title_col_name": {
+                "type": "string",
+                "doc": "The column that contains publication names or ref/id."
+            },
+            "corpus_table": {
+                "type": "table",
+                "doc": "The corpus table for which the distribution over time is needed."
+            }
+        }
+
+    def create_outputs_schema(self):
+        return {
+            "dist_table": {
+                "type": "table",
+                "doc": "The aggregated data table."
+            }
+        }
+
+    def process(self, inputs, outputs) -> None:
+
+        agg = inputs.get_value_obj("periodicity").data
+        title_col = inputs.get_value_obj("title_col_name").data
+        time_col = inputs.get_value_obj("date_col_name").data
+        
+        table_obj = inputs.get_value_obj("corpus_table")
+        
+        sources = pl.from_arrow(table_obj.data.arrow_table)
+
+        sources = sources.with_columns(pl.col(time_col).str.strptime(pl.Date, "%Y-%m-%d"))
+
```

Comment:
As I've said above, there should be some sort of assurance that if the 'agg' value is none of the 3 you use, some sort of error is thrown.

The sql queries look all right at first glance, and I guess it'd be fairly obvious fairly quickly if they were wrong, so I haven't tried to run them myself.

Reply:
This has now been addressed.